### PR TITLE
made utils and views python 3 compatible

### PR DESCRIPTION
**Pull request by krish8664**
- also corrected utils.py doctest to support python 3 as well as python 2

### CODE REVIEW

File: coaster/utils.py [hunk 1]
```diff
@@ -31,11 +34,23 @@
...
+    return r'[\t +!#$%&()*\-/<=>?@\[\\\]^_{|}:;,.…‒–—―«»]+'
+
+def punctuation_re_py2():
+    return six.text_type(r'[\t +!#$%&()*\-/<=>?@\[\\\]^_{|}:;,.…‒–—―«»]+', 'utf-8')
+
+_strip_re = re.compile(strip_re_py3() if six.PY3 else strip_re_py2())
+_punctuation_re = re.compile(punctuation_re_py3() if six.PY3 else punctuation_re_py2())
+_username_valid_re = re.compile(str('^[a-z0-9]([a-z0-9-]*[a-z0-9])?$'))
+_ipv4_re = re.compile(str('^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$'))
+_tag_re = re.compile(str('<.*?>'))
```
Comment (jace) [inline coaster/utils.py]: Do you need these to be functions? The `ur` prefix works in Py3 as well, right?

Comment (krish8664) [inline coaster/utils.py]: Sadly, no. u'' and r'' are accepted individually but there is no expression ur'' in Py3. It's a syntax error. I tried all variants to make to work in both Py3 and Py2. I couldn't even give it in a simple if - else. So the function, the only other option was try - except, but I thought this was a better approach.

Comment (jace) [inline coaster/utils.py]: But having `ur` in the code isn't a syntax error if the line isn't executed? How curious. I suggest avoiding the named functions then and putting in `lambda` functions where the function calls are.

Comment (krish8664) [inline coaster/utils.py]: That's what i thought, putting it in six.PY condition should only execute the code that satisfy that condition, but to my surprise it kept throwing error even though ideally that portion of the code shouldn't be run.

like if i replace my named function with the corresponding return statements in a simple if-else, it was still throwing syntax error. I'm guessing the lambda functions won't work either. but worth a try.

Comment (jace) [inline coaster/utils.py]: If it helps, we don't need to use raw strings. I just added that so there's less confusion about `\` characters in the string. Just escape those out and use a non-raw string.

Comment (krish8664) [inline coaster/utils.py]: Switched it to lambda function, but without using `ur''` as that still throws error. The code is a little bit cleaner now.

**Comment by jace** [comment]
Committed in ff7306006dae94078672403bba6f123805bd8b6e via #128.

**Pull request closed without merging**